### PR TITLE
Add cargo docs to public API

### PR DESCRIPTION
**Pull request by allonsy**

### CODE REVIEW

File: src/color.rs [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+/// A Color can be an RGB value, random, or the terminal's default color.
 #[derive(Clone, PartialEq, Debug)]
 pub enum Color {
+    /// Specify a specifc color using RGB values. Also see the equivalent [Color::rgb] function for the equivalent constructor.
     RGB(u8, u8, u8),
+    /// Represents a random color that will be determined at runtime.
+    /// NOTE: This isn't true random. The system looks at all the commands with the [Color::Random] variant and chooses semi-random colors
+    /// trying to maximize the distance (on the color wheel) between each command so each color is as visually distinct as possible.
+    /// This is to avoid cases where two commands have similar colors and it is hard to differentiate them due to random coincidence.
+    /// If you wish to have true random colors, you can either manually set RGB values manually or use the [Color::true_random] function.
```
Comment (kevgo) [inline src/color.rs:16]: double "manually" in this sentence

File: src/channel_api.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+/// This is the lowest-level of the three apis.
+/// It returns a channel which allows the caller to process all command output manually.
+/// This API provides the most flexibility at the cost of the most work to the user.
+/// ## Example:
+/// ```
+/// use currant::ChannelCommand;
+/// use currant::Command;
+/// use currant::OutputMessagePayload;
+/// use currant::Runner;
```
Comment (kevgo) [inline src/channel_api.rs:12]: merge these 4 lines?

**Review by kevgo — COMMENTED** [review]
Very cool, thanks for doing this!

File: src/channel_api.rs (hunk 2 above)
Comment (allonsy) [reply src/channel_api.rs:12]: implemented in #29

**Pull request merged**